autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/sync/CredentialsRateLimitView.kt

```diff
@@ -0,0 +1,121 @@
+import com.duckduckgo.autofill.impl.R
+import com.duckduckgo.autofill.impl.databinding.ViewCredentialsRateLimitWarningBinding
+import com.duckduckgo.autofill.sync.CredentialsRateLimitViewModel.Command
+import com.duckduckgo.autofill.sync.CredentialsRateLimitViewModel.Command.NavigateToCredentials
+import com.duckduckgo.autofill.sync.CredentialsRateLimitViewModel.ViewState
+import com.duckduckgo.common.ui.viewbinding.viewBinding
+import com.duckduckgo.common.utils.ConflatedJob
+import com.duckduckgo.di.scopes.ViewScope
+import com.duckduckgo.navigation.api.GlobalActivityStarter
+import dagger.android.support.AndroidSupportInjection
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@InjectWith(ViewScope::class)
+class CredentialsRateLimitView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+) : FrameLayout(context, attrs, defStyle) {
+
+    @Inject
+    lateinit var globalActivityStarter: GlobalActivityStarter
+
+    @Inject
+    lateinit var viewModelFactory: CredentialsRateLimitViewModel.Factory
+
+    private var coroutineScope: CoroutineScope? = null
+
+    private var job: ConflatedJob = ConflatedJob()
+
+    private val binding: ViewCredentialsRateLimitWarningBinding by viewBinding()
+
+    private val viewModel: CredentialsRateLimitViewModel by lazy {
+        ViewModelProvider(findViewTreeViewModelStoreOwner()!!, viewModelFactory)[CredentialsRateLimitViewModel::class.java]
+    }
+
+    override fun onAttachedToWindow() {
+        AndroidSupportInjection.inject(this)
+        super.onAttachedToWindow()
+
+        ViewTreeLifecycleOwner.get(this)?.lifecycle?.addObserver(viewModel)
+
+        configureViewListeners()
+
+        @SuppressLint("NoHardcodedCoroutineDispatcher")
+        coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
+
+        viewModel.viewState()
+            .onEach { render(it) }
+            .launchIn(coroutineScope!!)
+
+        job += viewModel.commands()
+            .onEach { processCommands(it) }
+            .launchIn(coroutineScope!!)
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        ViewTreeLifecycleOwner.get(this)?.lifecycle?.removeObserver(viewModel)
+        coroutineScope?.cancel()
+        job.cancel()
+        coroutineScope = null
+    }
+
+    private fun processCommands(command: Command) {
+        when (command) {
+            NavigateToCredentials -> navigateToCredentials()
+        }
+    }
+
+    private fun render(viewState: ViewState) {
+        this.isVisible = viewState.warningVisible
+    }
+
+    private fun configureViewListeners() {
+        binding.credentialsRateLimitWarning.setClickableLink(
+            "manage_logins",
```

Comment:
should this be a const defined somewhere? does it affect translations?

Reply:
Moved it into a const. It won't be translated as it's an annotation. 👍